src/poke_env/environment/abstract_battle.py

```diff
@@ -210,6 +210,54 @@ def get_pokemon(
     def _clear_all_boosts(self):  # pragma: no cover
         pass
 
+    def _check_damage_message_for_item(self, split_message: List[str]):
+        # Catches when a side takes damage from the opponent's item
+        # The item belongs to the side not taking damage
+        # Example:
+        #   |-damage|p2a: Archeops|88/100|[from] item: Rocky Helmet|[of] p1a: Ferrothorn
+        if (
+            len(split_message) == 6
+            and split_message[4].startswith("[from] item:")
+            and split_message[5].startswith("[of]")
+        ):
+            item = split_message[4].split("item:")[-1]
+            pkmn = split_message[5].split("[of]")[-1].strip()
+            self.get_pokemon(pkmn)._item = to_id_str(item)
```

Comment:
`Pokemon` objects have a setter method associated with the `item` property - could you do

```python
self.get_pokemon(pkmn).item = to_id_str(item)
```
instead? For now it's strictly equivalent, but if might be used for other callbacks at some point.